storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -330,7 +330,11 @@ public static Map readCommandLineOpts() {
         Map ret = new HashMap();
         String commandOptions = System.getProperty("storm.options");
         if (commandOptions != null) {
-            String[] configs = commandOptions.split(",");
+            /*
+             below regex uses negative lookahead to not split in the middle of
+             json objects '{}' or json arrays '[]'
+              */
+            String[] configs = commandOptions.split(",(?![^\\[\\]{}]*(]|}))");
```

Comment:
The regex works only for valid JSON strings.

With the current storm.cmd script, valid json strings were not parsed properly so one could not pass a list (for e.g. drpc.servers) from windows. I couldn't figure out a way to url-encode from storm.cmd so I came up with the regex that works for valid JSONs. 

The change should not affect the parsing of the storm.options sent via the python script since the url-encoded values would not contain `[, {, ] or }` and will not be touched by the regex.

Agree this is not the ideal fix that handles all the cases but lets one pass valid json in storm.cmd in windows which is currently broken. If there are any alternatives to get the options working in windows l can try to explore.